blockchain/state_migration.go

```diff
@@ -100,12 +100,12 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
 
 	start := time.Now()
 
-	srcCachedDB := bc.StateCache().TrieDB()
-	targetDB := statedb.NewDatabase(&stateTrieMigrationDB{bc.db})
+	srcState := bc.StateCache()
+	targetState := state.NewDatabase(&stateTrieMigrationDB{bc.db})
```

Comment:
Why did you change `statedb.NewDatabase` to `state.NetDatabase`?

Reply:
I changed it for `CheckStateConsistency` use a state with fast cache.